judge/views/problem_manage.py

```diff
@@ -82,15 +82,18 @@ def perform_action(self):
         except ValueError:
             return HttpResponseBadRequest()
 
-        return self.generate_response(id_range, languages, self.request.POST.getlist('result'))
+        archive_locked = self.request.user.has_perm('judge.archive_submission') and \
```

Comment:
I think having the archiving by default when locked would be better when the user doesn't have this permission. For example, some admin locks some submissions for a problem. Then, a problem setter wants to rejudge all submissions on their problem. Locked submissions should be automatically archived in my opinion.

This use case is probably more common than a problem setter _not_ wanting to archive locked submissions.